### PR TITLE
Support running unit and integration tests

### PR DESCRIPTION
**Pull request by sbryzak**
Add basic support for executing unit and integration tests

Fixes https://issues.redhat.com/projects/CRT/issues/CRT-1626?filter=myopenissues

### CODE REVIEW

File: test/require.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	// specify that unit tests shall be run. Unless this environment variable
+	// is explicitly set to evaluate to false ("0", "no", or "false"), unit
+	// tests are executed all the time.
+	UnitTest = "AUTH_RESOURCE_UNIT_TEST"
+	// Database refers to the name of the environment variable that is used to
+	// specify that test can be run that require a database.
+	Database = "AUTH_RESOURCE_DATABASE"
+	// Remote refers to the name of the environment variable that is used to
+	// specify that test can be run that require availability of some remote servers such as Keycloak.
+	Remote = "AUTH_RESOURCE_REMOTE"
```
Comment (alexeykazakov) [inline test/require.go]: I don't think we need this mode. Let's drop `Remote`.

File: go.mod [hunk 2]
```diff
@@ -4,53 +4,160 @@ go 1.19
...
+	github.com/containerd/containerd v1.6.8 // indirect
+	github.com/containerd/continuity v0.3.0 // indirect
+	github.com/containerd/ttrpc v1.1.0 // indirect
+	github.com/containerd/typeurl v1.0.2 // indirect
+	github.com/davecgh/go-spew v1.1.1 // indirect
 	github.com/dimfeld/httptreemux/v5 v5.2.2 // indirect
-	github.com/fsnotify/fsnotify v1.6.0 // indirect
+	github.com/distribution/distribution/v3 v3.0.0-20220907155224-78b9c98c5c31 // indirect
+	github.com/docker/buildx v0.9.1 // indirect
+	github.com/docker/cli v20.10.19+incompatible // indirect
```
Comment (alexeykazakov) [inline go.mod:34]: I wonder what requires docker modules?

File: test/require.go (hunk 1 above)
Comment (sbryzak) [reply test/require.go]: Removed.

File: go.mod (hunk 2 above)
Comment (sbryzak) [reply go.mod:34]: This is a requirement for testcontainers.

**Review by alexeykazakov — APPROVED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# syntax=docker/dockerfile:1
+FROM golang:1.19-buster
```
Comment (xcoulon) [inline Dockerfile:2]: we build everything else with Go 1.18. Do we really need Go 1.19 here?

File: pkg/errors/errors_test.go [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+	"github.com/codeready-toolchain/sandbox-auth/test"
+	errs "github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"testing"
+)
+
+func TestNewInternalError(t *testing.T) {
+	t.Parallel()
```
Comment (xcoulon) [inline pkg/errors/errors_test.go:15]: why do we need to provide our own `errors` package?

File: pkg/errors/errors_test.go [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+	err := errors.NewInternalError(errs.New("system disk could not be read"))
+
+	// not sure what assertion to do here.
+	t.Log(err)
+}
+
+func TestNewConversionError(t *testing.T) {
+	t.Parallel()
+	test.Require(t, test.UnitTest)
+	err := errors.NewConversionError("Couldn't convert workitem")
```
Comment (xcoulon) [inline pkg/errors/errors_test.go:26]: is this "workitem" related to fabric8-wit?

File: pkg/application/factory/manager/manager.go [hunk 6]
```diff
@@ -1,17 +1,15 @@
 package factory
 
-import (
-	context2 "github.com/codeready-toolchain/sandbox-auth/pkg/application/service/context"
-)
-
 type Manager struct {
-	contextProducer context2.ServiceContextProducer
+	//contextProducer context2.ServiceContextProducer
```
Comment (xcoulon) [inline pkg/application/factory/manager/manager.go:4]: we don't need to keep commented out code

**Pull request merged**